addon/components/bs-link-to.js

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable ember/classic-decorator-no-classic-methods */
+import Component from '@ember/component';
+import { tagName } from '@ember-decorators/component';
+import { computed } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { assert } from '@ember/debug';
+
+/**
+ This is largely copied from Ember.LinkComponent. It is used as extending from Ember.LinkComponent has been deprecated.
+ We need this to
+ * register ourselves to a parent component that needs to know `active` state due to Bootstrap markup requirements, see Nav/LinkTo
+ * continue supporting positional params until we can remove support
+
+ @class LinkComponent
+ @namespace Components
+ @extends Component
+ @private
+*/
+@tagName('')
+class LinkComponent extends Component {
+  // We still need to use the private service, so this component can react to changes of `currentState`, e.g. changing
+  // the model while the route in unchanged.
+  // eslint-disable-next-line ember/no-private-routing-service
+  @service('-routing')
+  _routing;
+
+  @computed('models', 'query', 'route', '_routing.currentState', 'current-when')
+  get active() {
+    let state = this._routing.currentState;
+
+    if (state) {
+      return this._isActive(state);
+    } else {
+      return false;
+    }
+  }
+
+  @computed('model', 'models')
+  get _models() {
+    let { model, models } = this;
+
+    if (model !== undefined) {
+      return [model];
+    } else if (models !== undefined) {
+      assert('The `@models` argument must be an array.', Array.isArray(models));
+      return models;
+    } else {
+      return [];
+    }
+  }
+
+  _isActive(routerState) {
+    let currentWhen = this['current-when'];
+
+    if (typeof currentWhen === 'boolean') {
+      return currentWhen;
+    }
+
+    let { _models: models, _routing: routing } = this;
+
+    if (typeof currentWhen === 'string') {
+      return currentWhen
+        .split(' ')
+        .some((route) => routing.isActiveForRoute(models, undefined, this._namespaceRoute(route), routerState));
```

Comment:
Ah, shit. I haven't copied `this._namespaceRoute()` from the original LinkComponent, which seems to be needed for the `current-when`. This would require even more stupid copy-pasting.

I wonder if we need to support this: `current-when` is nowhere mentioned in our API docs, it's just the fact that we documented the component to extend from `Ember.LinkComponent` (shouldn't have done this probably!?). So can we drop this nasty stuff without making this a breaking change? 🤔